src/components/cluster-deployments/ClusterDeploymentWizard.tsx

```diff
@@ -0,0 +1,203 @@
+
+          // TODO(mlibra): InstallEnv should be patched for the ClusterDeployment reference
+        } catch (e) {
+          // A string-only is expected. Or change the ClusterDeploymentDetails in the assisted-ui-lib
+          throw `Failed to cretate the ClusterDeployment or gentClusterInstall resource: ${e.message}`;
+        }
+      },
+    [setClusterDeployment, setAgentClusterInstall, namespace],
+  );
+
+  const onClusterDetailsUpdate = React.useMemo(
+    () => async (values: ClusterDeploymentDetailsValues) => {
+      // do we need to re-query the ClusterDeployment??
+      try {
+        const clusterDeploymentPatches = [];
+        const agentClusterInstallPatches = [];
+
+        // pullSecret, highAvailabilityMode, useRedHatDnsService can not be changed
+
+        /* TODO(mlibra): Uncomment once we can update the ClusterDeployment resource
+      Issue: Failed to patch the ClusterDeployment resource: admission webhook "clusterdeploymentvalidators.admission.hive.openshift.io" denied
+      the request: Attempted to change ClusterDeployment.Spec which is immutable 
+      except for CertificateBundles,ClusterMetadata,ControlPlaneConfig,Ingress,Installed,PreserveOnDelete,ClusterPoolRef,PowerState,HibernateAfter,InstallAttemptsLimit,MachineManagement fields.
+      Unsupported change: ClusterName: (mlibra-05 => mlibra-07)
+
+        appendPatch(clusterDeploymentPatches, '/spec/clusterName', values.name, clusterDeployment.spec?.clusterName);
+        appendPatch(
+          clusterDeploymentPatches,
+          '/spec/baseDomain',
+          values.baseDnsDomain,
+          clusterDeployment.spec?.baseDomain,
+        );
+        */
+
+        // TODO(mlibra): set values.openshiftVersion to agentClusterInstall?.spec?.imageSetRef?.name
+        appendPatch(
+          agentClusterInstallPatches,
+          '/spec/imageSetRef/name',
+          values.openshiftVersion,
+          agentClusterInstall.spec.imageSetRef?.name,
+        );
+
+        if (clusterDeploymentPatches.length > 0) {
+          await k8sPatch(clusterDeploymentModel, clusterDeployment, clusterDeploymentPatches);
+        }
+        if (agentClusterInstallPatches.length > 0) {
+          await k8sPatch(agentClusterInstallModel, agentClusterInstall, agentClusterInstallPatches);
+        }
+      } catch (e) {
+        throw `Failed to patch the ClusterDeployment or AgentClusterInstall resource: ${e.message}`;
+      }
+    },
+    [],
+  );
+
+  const onSaveDetails = React.useMemo(
+    () => async (values: ClusterDeploymentDetailsValues) => {
+      if (clusterDeployment) {
+        // we have already either queried (the Edit flow) or created it
+        await onClusterDetailsUpdate(values);
+      } else {
+        await onClusterCreate(values);
+      }
+    },
+    [onClusterCreate, onClusterDetailsUpdate, clusterDeployment],
+  );
+
+  const onClose = () => {
+    history.push(`/k8s/ns/${namespace || 'all-namespaces'}/${ClusterDeploymentKind}`);
```

Comment:
the `all-namespace` path is wrong. Should be just `k8s/all-namespaces` not `k8s/ns/all-namespaces`

Reply:
thanks, done